src/generators/pm/utils.ts

```diff
@@ -0,0 +1,21 @@
+import { readPackageJSON } from "pkg-types";
+
+export async function inferPackageName(dir: string) {
+  const pkgName = await readPackageJSON(dir)
+    .then((pkg) => pkg?.name)
+    .catch(() => undefined);
+  return pkgName || process.env.npm_package_name;
+}
```

Comment:
I was going to use this util for `badges` maybe it util root vs `pm/util.ts`

Reply:
refactored 👍🏼